packages/compass-indexes/src/components/create-search-index-modal/index.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useCallback, useState } from 'react';
+import { useTrackOnChange } from '@mongodb-js/compass-logging';
+import {
+  Modal,
+  ModalFooter,
+  ModalHeader,
+  ModalBody,
+  TextInput,
+  Label,
+  spacing,
+  css,
+  HorizontalRule,
+  Subtitle,
+  Button,
+  Link,
+  Icon,
+  WarningSummary,
+  ErrorSummary,
+} from '@mongodb-js/compass-components';
+import { closeModal, saveIndex } from '../../modules/search-indexes';
+import { connect } from 'react-redux';
+import { CodemirrorMultilineEditor } from '@mongodb-js/compass-editor';
+import type { RootState } from '../../modules';
+import _parseShellBSON, { ParseMode } from 'ejson-shell-parser';
+import type { Document } from 'mongodb';
+
+// Copied from packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/utils.ts
+function parseShellBSON(source: string): Document[] {
+  const parsed = _parseShellBSON(source, { mode: ParseMode.Loose });
+  if (!parsed || typeof parsed !== 'object') {
+    // XXX(COMPASS-5689): We've hit the condition in
+    // https://github.com/mongodb-js/ejson-shell-parser/blob/c9c0145ababae52536ccd2244ac2ad01a4bbdef3/src/index.ts#L36
+    throw new Error('The provided index definition is invalid.');
+  }
+  return parsed;
+}
+
+export const DEFAULT_INDEX_DEFINITION = `{
+  "mappings": {
+    "dynamic": true
+  }
+}`;
+
+const bodyGapStyles = css({
+  marginTop: spacing[3],
+});
```

Comment:
Instead of adding this to every dom element that needs a gap added, you can use gap with flex or grid on a parent to make sure that all elements are correctly aligned and if we add new ones or move them around the gap stays the same